packages/@aws-cdk/core/lib/cfn-resource.ts

```diff
@@ -289,6 +300,45 @@ export class CfnResource extends CfnRefElement {
     addDependency(this, target, `"${Node.of(this).path}" depends on "${Node.of(target).path}"`);
   }
 
+  /**
+   * Indicates that this resource no longer depends on another resource.
+   *
+   * This can be used for resources across stacks (including nested stacks)
+   * and the dependency will automatically be removed from the relevant scope.
+   */
+  public removeDependency(target: CfnResource) : void {
+    // skip this dependency if the target is not part of the output
+    if (!target.shouldSynthesize()) {
+      return;
+    }
+
+    removeDependency(this, target, `"${Node.of(this).path}" depends on "${Node.of(target).path}"`);
```

Comment:
Why does this need a reason description at all? A source and a target should be enough.

Reply:
This provides an argument for the reason that is completely different from the default value of `removeDependency()`'s parameter, and I still don't see why either of them exist.

Reply:
Yeah the reason description definitely doesn't need to exist at all and I'm happy to remove. I added that because, in the existing internal api, there was that `reason` string, which seemed to be diagnostic in nature, so I kept a string in the `DependencyReason` as an attempt to preserve that behavior.